nni/compression/pytorch/speedup/infer_shape.py

```diff
@@ -891,23 +891,18 @@ def convert_to_coarse_mask(mask, dim=0):
         sum_idx = (1, 2, 3) if dim == 0 else (0, 2, 3)
         index = torch.nonzero(weight_mask.abs().sum(
             sum_idx) != 0, as_tuple=True)[0]
-        if len(index) == weight_mask.shape[dim]:  # full mask
-            index = None
 
-        if index is None:
-            return None, None, None
```

Comment:
so index cannot be None, why?

Reply:
if `index` is None, then we will skip this layer, it's fine when this mask is a fine-grained mask, however, this may also skip the all-ones mask added by the CatMaskPadding which will lead to a shape mis-match.